modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AspNetCoreServerCodegen.java

```diff
@@ -482,4 +513,61 @@ private void setOperationIsAsync() {
             additionalProperties.put(OPERATION_IS_ASYNC, operationIsAsync);
         }
     }
+
+    private void setIsFramework() {
+        if (aspnetCoreVersion.getOptValue().startsWith("3.")) {
+            // default, do nothing
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so changing  to use frameworkReference instead of packageReference ");
+            isFramework = true;
+            additionalProperties.put(IS_FRAMEWORK, isFramework);
+        } else {
+            if (additionalProperties.containsKey(IS_FRAMEWORK)) {
+                isFramework = convertPropertyToBooleanAndWriteBack(IS_FRAMEWORK);
+            } else {
+                additionalProperties.put(IS_FRAMEWORK, isFramework);
+            }
+        }
+
+    }
+
+    private void setUseNewtonsoft() {
+        if (aspnetCoreVersion.getOptValue().startsWith("2.")) {
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so staying on default json library.");
+            useNewtonsoft = false;
+            additionalProperties.put(USE_NEWTONSOFT, useNewtonsoft);
+        } else {
+            if (additionalProperties.containsKey(USE_NEWTONSOFT)) {
+                useNewtonsoft = convertPropertyToBooleanAndWriteBack(USE_NEWTONSOFT);
+            } else {
+                additionalProperties.put(USE_NEWTONSOFT, useNewtonsoft);
+            }
+        }
+    }
+
+    private void setUseEndpointRouting() {
+        if (aspnetCoreVersion.getOptValue().startsWith("3.")) {
+            LOGGER.warn("ASP.NET core version is " + aspnetCoreVersion.getOptValue() + " so switching to old style endpoint routing.");
+            useDefaultRoutng = false;
+            additionalProperties.put(USE_DEFAULT_ROUTING, useDefaultRoutng);
+        } else {
+            if (additionalProperties.containsKey(USE_DEFAULT_ROUTING)) {
+                useDefaultRoutng = convertPropertyToBooleanAndWriteBack(USE_DEFAULT_ROUTING);
+            } else {
+                additionalProperties.put(USE_DEFAULT_ROUTING, useDefaultRoutng);
+            }
+        }
+    }
+
+    private void setSwashbuckleVersion() {
+        setCliOption(swashbuckleVersion);
```

Comment:
Long comment, but minor (can be addressed later).

Only the CLI should be setting CLI options. We should read options once (in `processOpts`), then have nothing to do with CLIOption types. That is, we should load into fields and work from those fields directly. I'm concerned that making this method side-effecting as it is and assuming options parsing within the generator, things can become out of sync. 

For example, a user can pass `--additional-properties=swashbuckleVersion=3.0.5". Although this isn't defined in the enum of available versions, this is a valid use case that we allow for properties defined in `additionalProperties`; this is why you often see the guard that we only set the value if it doesn't already exist.  `CodegenConfigurator` also allows for JSON deserialization into the `CodegenConfigurator` type (to load these values from a config), which sorta bypasses both the cli options and additional properties (see [CodegenConfigurator](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/config/CodegenConfigurator.java#L634)).

In all of these cases, this method will reset the user-provided value to whatever is originally hardcoded as the swashbuckleVersion default within the generator.

Reply:
The other choice here is to check the version of Swashbuckle is valid for the ASP.NET Core chosen and  throw an error if the version is not valid. 

In this particular case I think you can use v 4.0.0 with version 2.x and 3.x but you cannot use v3.0.0 with 3.x and later.

What are you thoughts here - error or warn and correct - not sure what the general policy for this is.

Reply:
I think my preference would be to set a default if one doesn't exist, and warn if it's something we don't expect will work. We'll probably want to be clear that anything below 4.0.0 _may_ not work, but trust that the user understands the versioning.  For example, someone might have a fork of Swashbuckle which does work, and they could reference and add the file with the reference to `.openapi-generator-ignore`. In that case, a warn+correct would be unexpected if the value was explicitly provided.